doc: Doc updates related to Python deps (#12091)
nrfconnect/sdk-nrf: doc/nrf/installation/installing.rst

In this case, *next_revision* can be either a SHA (for example, ``224bee9055d986fe2677149b8cbda0ff10650a6e``), a branch, or a tag name.
You can use any revision of the sdk-nrf repository that is available on your local file system.

.. _additional_deps:

.. rst-class:: numbered-step

Install additional Python dependencies
**************************************

The |NCS| requires additional Python packages to be installed.

Use the following commands to install the requirements for each repository.

.. tabs::

   .. group-tab:: Windows

      Enter the following command in a command-line window in the :file:`ncs` folder:

        .. parsed-literal::
           :class: highlight

           pip3 install -r zephyr/scripts/requirements.txt
           pip3 install -r nrf/scripts/requirements.txt

>            pip3 install -r bootloader/mcuboot/scripts/requirements.txt

Reviewer: I think this is still required to e.g. build Zephyr or mcuboot successfully

The |NCS| requires additional Python packages to be installed.

.. note::

>     It is easy to run into Python package incompatibilities when installing dependencies at a system or user level.

Reviewer 2: ```suggestion
    You might run into Python package incompatibilities when installing dependencies at a system or user level.
```

    For this reason, it is suggested to use `Python virtual environments`_.

Use the following commands to install the requirements for each repository.

.. tabs::

   .. group-tab:: Windows

>       1. Use pip command to install Python ``venv`` package:

Reviewer 3: Should it be "the Python ``venv`` package" with the definitive article?

            :class: highlight

            pip3 install virtualenv

      #. Create a new virtual environment:

         .. parsed-literal::
            :class: highlight

            cd %HOMEPATH%
            python -m venv ncs\\.venv

      #. Activate the virtual environment:

         .. parsed-literal::
            :class: highlight

            :: cmd.exe
            ncs\\.venv\\Scripts\\activate.bat
            :: PowerShell
            ncs\\.venv\\Scripts\\Activate.ps1

         Once activated, your shell will be prefixed with ``(.venv)``.
         The virtual environment can be deactivated at any time by running ``deactivate``.

         .. note::

            Remember to activate the virtual environment every time you start working.

      #. Enter the following commands in a command line window in the :file:`ncs` folder:

         .. parsed-literal::
            :class: highlight

            pip3 install -r  %HOMEPATH%\\ncs\\zephyr\\scripts\\requirements.txt
            pip3 install -r %HOMEPATH%\\ncs\\nrf\\scripts\\requirements.txt
            pip3 install -r %HOMEPATH%\\ncs\\bootloader\\mcuboot\\scripts\\requirements.txt

   .. group-tab:: Linux

>       1. Use apt to install Python ``venv`` package:

Reviewer 2: ```suggestion
      1. Use the following command to install Python ``venv`` package:
```

         Once activated, your shell will be prefixed with ``(.venv)``.
         The virtual environment can be deactivated at any time by running ``deactivate``.

         .. note::

            Remember to activate the virtual environment every time you start working.

      #.  Enter the following commands in a terminal window in the :file:`ncs` folder:

          .. parsed-literal::
             :class: highlight

             pip3 install --user -r zephyr/scripts/requirements.txt
             pip3 install --user -r nrf/scripts/requirements.txt
             pip3 install --user -r bootloader/mcuboot/scripts/requirements.txt

   .. group-tab:: macOS

      1. Use apt to install Python ``venv`` package:

         .. parsed-literal::
            :class: highlight

            sudo apt install python3-venv

      #. Create a new virtual environment:

         .. parsed-literal::
            :class: highlight

            python3 -m venv ~/ncs/.venv

      #. Activate the virtual environment:

         .. parsed-literal::
            :class: highlight

            source ~/ncs/.venv/bin/activate

         Once activated, your shell will be prefixed with ``(.venv)``.

>          The virtual environment can be deactivated at any time by running ``deactivate``.

Reviewer 2: ```suggestion
         You can deactivate the virtual environment at any time by running ``deactivate``.
```
